plutus-core/plutus-core/src/PlutusCore/Core/Instance/Pretty/Readable.hs

```diff
@@ -44,10 +44,8 @@ instance PrettyBy (PrettyConfigReadable configName) (Kind a) where
         Type{}          -> "*"
         KindArrow _ k l -> k `arrowPrettyM` l
 
-instance
-        ( PrettyReadableBy configName tyname
-        , PrettyParens (SomeTypeIn uni)
-        ) => PrettyBy (PrettyConfigReadable configName) (Type tyname uni a) where
+instance (PrettyReadableBy configName tyname, PrettyParens (SomeTypeIn uni)) =>
```

Comment:
> It reads weirdly to me. 

Suggestions are welcome, but it's literally for rendering the thing with a minimum amount of parens. Which means different things in the classic and the readable pretty-printer. It doesn't have much to do with the "readable" syntax, apart from being implemented using the same machinery and fitting into the readable pretty-printing as well as it fits into classic pretty-printing. To give you an example, in the classic syntax `(list data)` is always rendered in parens and in the readable syntax whether to add parens or not depends on what's around that type.

> I would expect something like `PrettyReadableBy configName (SomeTypeIn uni)`

This was my first thought too, you can read a previous [comment](https://github.com/input-output-hk/plutus/pull/5291#discussion_r1184835771) to see why I don't believe it would be a good idea.

> Indeed, it looks like you meant to get rid of it, should be removed entirely?

That wasn't my intention at all. In the previous PR I wanted to

1. teach the readable pretty-printing machinery how to render built-in types in the current context -- without breaking the classic pretty-printing
2. do that in a way that doesn't require two constraints (one for rendering built-in types using the classic syntax and one for rendering them using the readable syntax)

In this PR I'm only cleaning up constraints by introducing/moving/renaming/making use of type aliases without changing any logic whatsoever.

Reply:
> Yes, but that's like the only point of RenderContext -- to render things with minimal parens.

That's not at all obvious to me :/ I think I just don't actually understand the pretty-printing machinery...

Reply:
> That's not at all obvious to me :/ I think I just don't actually understand the pretty-printing machinery...

I have really good news for you then! That specific thing is well-documented and self-contained, do check out the [docs](https://github.com/input-output-hk/plutus/blob/db8bb46d921a6906741b36210e965465d4f539b4/prettyprinter-configurable/src/Text/Fixity/Internal.hs). It even describes a system that seems to provide better utility-to-complexity ratio than any other system I was able to find (I didn't implement it, because we didn't need it), so maybe you'll find it interesting enough for me to persuade you to write a tiny paper on it or something :)